### ISSUE TITLE
self.assertFailure return value should be waited for

### ISSUE THREAD

**Issue by tardyp** [buildbot/buildbot#4253]
as per https://twistedmatrix.com/trac/ticket/9505

self.assertFailure returns a deferred, that should be waited for in the test.

```
git grep -E '^ +self.assertFailure'
master/buildbot/test/unit/test_changes_github.py:        self.assertFailure(d, KeyError)
master/buildbot/test/unit/test_changes_p4poller.py:        self.assertFailure(d, P4PollerError)
master/buildbot/test/unit/test_db_buildsets.py:        self.assertFailure(d, AssertionError)
master/buildbot/test/unit/test_db_changesources.py:        self.assertFailure(d, changesources.ChangeSourceAlreadyClaimedError)
master/buildbot/test/unit/test_db_changesources.py:        self.assertFailure(d, changesources.ChangeSourceAlreadyClaimedError)
master/buildbot/test/unit/test_db_schedulers.py:        self.assertFailure(d, schedulers.SchedulerAlreadyClaimedError)
master/buildbot/test/unit/test_db_schedulers.py:        self.assertFailure(d, schedulers.SchedulerAlreadyClaimedError)
master/buildbot/test/unit/test_db_steps.py:        self.assertFailure(d, RuntimeError)
master/buildbot/test/unit/test_interpolate_secrets.py:        self.assertFailure(self.build.render(command), defer.FirstError)
master/buildbot/test/unit/test_interpolate_secrets.py:        self.assertFailure(self.build.render(command), defer.FirstError)
master/buildbot/test/unit/test_process_properties.py:        self.assertFailure(d, TypeError)
master/buildbot/test/unit/test_process_properties.py:        self.assertFailure(d, RuntimeError)
master/buildbot/test/unit/test_stats_service.py:        self.assertFailure(cap.consume(self.routingKey, self.get_dict(build)),
master/buildbot/test/unit/test_stats_service.py:        self.assertFailure(cap.consume(self.routingKey, self.get_dict(build)),
master/buildbot/test/unit/test_stats_service.py:        self.assertFailure(cap.consume(self.routingKey, self.get_dict(build)),
master/buildbot/test/unit/test_stats_service.py:        self.assertFailure(cap.consume(self.routingKey, self.get_dict(build)),
master/buildbot/test/unit/test_stats_service.py:        self.assertFailure(cap.consume(routingKey, msg), CaptureCallbackError)
master/buildbot/test/unit/test_util_codebase.py:        self.assertFailure(d, KeyError)
master/buildbot/test/unit/test_util_lru.py:            self.assertFailure(d, RuntimeError)
master/buildbot/test/unit/test_util_misc.py:        self.assertFailure(d, RuntimeError)
master/buildbot/test/unit/test_util_misc.py:        self.assertFailure(d, defer.CancelledError)
master/buildbot/test/unit/test_util_misc.py:        self.assertFailure(d, defer.CancelledError)
master/buildbot/test/unit/test_worker_openstack.py:        self.assertFailure(bs.start_instance(self.build), ValueError)
master/buildbot/test/unit/test_www_auth.py:        self.assertFailure(defer_bad, UnauthorizedLogin)
worker/buildbot_worker/test/unit/test_commands_transfer.py:        self.assertFailure(d, RuntimeError)
worker/buildbot_worker/test/unit/test_commands_transfer.py:        self.assertFailure(d, RuntimeError)
```

**Comment by mokibit** [comment]
Working on this one.

**Comment by yan12125** [comment]
@tardyp I consider this issue not fully fixed as the workaround 7fa7259 is still necessary. Should I open a new issue?
